src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/PackagePropertyPage/PackageIconValueProvider.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties.Package
+{
+    [ExportInterceptingPropertyValueProvider(PackageIconPropertyName, ExportInterceptingPropertyValueProviderFile.ProjectFile)]
+    internal sealed class PackageIconValueProvider : InterceptingPropertyValueProviderBase
```

Comment:
It would be helpful for future generations if we added a comment explaining the intended final state of the project.

I wouldn't worry about making it an XML doc. Just a multi-line `//` comment within the class showing the MSBuild property and items at play here.

Reply:
Done. Added a description similar to a shorter version of the description of this PR, and included example XML output.